Source/Core/Core/DSP/DSPHWInterface.cpp

```diff
@@ -23,349 +21,346 @@
   case DSP_ACCELERATOR:  // ADPCM Accelerator reads
-    return g_dsp.accelerator->Read(reinterpret_cast<s16*>(&g_dsp.ifx_regs[DSP_COEF_A1_0]));
+    return accelerator->Read(reinterpret_cast<s16*>(&ifx_regs[DSP_COEF_A1_0]));
   case DSP_ACDATA1:  // Accelerator reads (Zelda type) - "UnkZelda"
-    return g_dsp.accelerator->ReadD3();
+    return accelerator->ReadD3();
 
   default:
   {
-    const u16 ifx_reg = g_dsp.ifx_regs[addr & 0xFF];
+    const u16 ifx_reg = ifx_regs[address & 0xFF];
 
-    if ((addr & 0xff) >= 0xa0)
+    if ((address & 0xff) >= 0xa0)
     {
-      const u32 index = (addr & 0xFF) - 0xa0;
+      const u32 index = (address & 0xFF) - 0xa0;
       const auto& label = pdlabels[index];
 
       if (label.name && label.description)
       {
-        DEBUG_LOG_FMT(DSPLLE, "{:04x} MR {} ({:04x})", g_dsp.pc, label.name, ifx_reg);
+        DEBUG_LOG_FMT(DSPLLE, "{:04x} MR {} ({:04x})", pc, label.name, ifx_reg);
       }
       else
       {
-        ERROR_LOG_FMT(DSPLLE, "{:04x} MR {:04x} ({:04x})", g_dsp.pc, addr, ifx_reg);
+        ERROR_LOG_FMT(DSPLLE, "{:04x} MR {:04x} ({:04x})", pc, address, ifx_reg);
       }
     }
     else
     {
-      ERROR_LOG_FMT(DSPLLE, "{:04x} MR {:04x} ({:04x})", g_dsp.pc, addr, ifx_reg);
+      ERROR_LOG_FMT(DSPLLE, "{:04x} MR {:04x} ({:04x})", pc, address, ifx_reg);
     }
     return ifx_reg;
   }
   }
 }
 
-u16 gdsp_ifx_read(u16 addr)
+u16 SDSP::ReadIFX(u16 address)
 {
-  u16 retval = _gdsp_ifx_read(addr);
-  g_dsp_cap->LogIFXRead(addr, retval);
+  const u16 retval = ReadIFXImpl(address);
+  m_dsp_core.LogIFXRead(address, retval);
   return retval;
 }
 
-static const u8* gdsp_idma_in(u16 dsp_addr, u32 addr, u32 size)
+const u8* SDSP::IDMAIn(u16 dsp_addr, u32 addr, u32 size)
 {
-  Common::UnWriteProtectMemory(g_dsp.iram, DSP_IRAM_BYTE_SIZE, false);
-  Host::DMAToDSP(g_dsp.iram + dsp_addr / 2, addr, size);
-  Common::WriteProtectMemory(g_dsp.iram, DSP_IRAM_BYTE_SIZE, false);
+  Common::UnWriteProtectMemory(iram, DSP_IRAM_BYTE_SIZE, false);
+  Host::DMAToDSP(iram + dsp_addr / 2, addr, size);
+  Common::WriteProtectMemory(iram, DSP_IRAM_BYTE_SIZE, false);
 
-  Host::CodeLoaded(addr, size);
+  Host::CodeLoaded(m_dsp_core, addr, size);
   NOTICE_LOG_FMT(DSPLLE, "*** Copy new UCode from {:#010x} to {:#06x} (crc: {:#08x})", addr,
-                 dsp_addr, g_dsp.iram_crc);
+                 dsp_addr, iram_crc);
 
-  return reinterpret_cast<u8*>(g_dsp.iram) + dsp_addr;
+  return reinterpret_cast<const u8*>(iram) + dsp_addr;
 }
 
-static const u8* gdsp_idma_out(u16 dsp_addr, u32 addr, u32 size)
+const u8* SDSP::IDMAOut(u16 dsp_addr, u32 addr, u32 size)
 {
   ERROR_LOG_FMT(DSPLLE, "*** idma_out IRAM_DSP ({:#06x}) -> RAM ({:#010x}) : size ({:#010x})",
                 dsp_addr / 2, addr, size);
-
   return nullptr;
 }
 
 // TODO: These should eat clock cycles.
-static const u8* gdsp_ddma_in(u16 dsp_addr, u32 addr, u32 size)
+const u8* SDSP::DDMAIn(u16 dsp_addr, u32 addr, u32 size)
```

Comment:
Could some of these in / out functions be `const`?

Reply:
Yes, but I'd rather they be left non-const, given if the data DMA ones are made more accurate in the future (by eating cycles), they'd need to become non-const anyways. That and the instruction DMA ones alter the write protections on the iram pointer. I could mark IDMAOut as const, but this would be an inconsistent interface.